android/app/src/main/java/org/fossasia/openevent/utils/NotificationUtil.java

```diff
@@ -37,12 +37,15 @@ public static Completable createNotification(Session session, Context context) {
             } else {
                 zonedDateTime.minusHours(10);
             }
-            Intent myIntent = new Intent(context, NotificationAlarmReceiver.class);
-            myIntent.putExtra(ConstantStrings.SESSION, session.getId());
-            PendingIntent pendingIntent = PendingIntent.getBroadcast(context, 0, myIntent, PendingIntent.FLAG_UPDATE_CURRENT);
-
-            AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
-            alarmManager.set(AlarmManager.RTC, zonedDateTime.toInstant().toEpochMilli(), pendingIntent);
+            // Checking if the event time is before the current time
+            if (zonedDateTime.isBefore(ZonedDateTime.now())) {
+                Intent myIntent = new Intent(context, NotificationAlarmReceiver.class);
+                myIntent.putExtra(ConstantStrings.SESSION, session.getId());
+                PendingIntent pendingIntent = PendingIntent.getBroadcast(context, 0, myIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+
+                AlarmManager alarmManager = (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
+                alarmManager.set(AlarmManager.RTC, zonedDateTime.toInstant().toEpochMilli(), pendingIntent);
+            }else Timber.i("Event is finished");
```

Comment:
Please add a space after curly bracket and wrap the logging statement in curly braces as well.

Also, use `Timber.d` instead as this is a debugging message, and for the message you may use

`Session is finished. Skipping showing notification` so that the developers are aware of the context :smile: